File: Array/Easy/2244. Minimum Rounds to Complete All Tasks.js
/**
 * @param {number[]} tasks
 * @return {number}
 */
var minimumRounds = function (tasks) {
	let obj = {};
	tasks.forEach((element) => {
		if (!obj[element]) {
			obj[element] = 1;
		} else {
			obj[element]++;
		}
    });
    let minRound = 0
    for (const key in obj) {
        let value = obj[key]
        if (value < 2) {
            return -1
        } else if(value === 3 || value === 2) {
            minRound++
        } else if (value > 3) {
            minRound += Math.ceil((value)/3)
        }
    }
	return minRound;
};
console.log(minimumRounds([2, 2, 3, 3, 2, 4, 4, 4, 4, 4]));

File: Array/Easy/2244. Minimum Rounds to Complete All Tasks.cpp
#include<iostream>
#include<vector>
#include <unordered_map>
using namespace std;
class Solution
{
public:
    // function to convert a number into its ceil form.

    int ceilTo(int x, int y)
    {
        if (x % y)
        {
            return (x / y) + 1;
        }

        return x / y;
    }

    int minimumRounds(vector<int> &tasks)
    {

        unordered_map<int, int> m;

        // inserting all the array elements into hashmap.
        for (auto x : tasks)
        {
            m[x]++;
        }

        int ans = 0; // variable to store total rounds required

        for (auto x : m)
        {
            if (x.second == 1)
                return -1;

            if (x.second > 3)
            {
                ans += ceilTo(x.second, 3);
            }
            else
            {
                ans += 1;
            }
        }
        // x.second means frequency

        return ans;
    }
};

int main(){
    Solution solution;
    vector<int> num = {2, 2,  3, 3, 3, 4, 4, 4, 4};
    int ans = solution.minimumRounds(num);
    cout << ans << "\n";
    return 0;
}